src/sandbox.rkt

```diff
@@ -35,176 +34,159 @@
 (define (get-cost test)
+  (setup-backend! test)
   (expr-cost (test-input test) (test-output-repr test)))
 
-
-;; Translates points from the API endpoint
-;; into the expected pcontext
-(define (compute-pcontexts pts+exs ctx)
-  (define output-repr (context-repr ctx))
-  (define var-reprs (context-var-reprs ctx))
-
-  (define-values (pts exs)
-    (for/lists (pts exs) ([entry (in-list pts+exs)])
-      (match-define (list pt ex) entry)
-      (values (map real->repr pt var-reprs) (real->repr ex output-repr))))
-
-  (define joint-pcontext (mk-pcontext pts exs))
-  (define-values (train-pcontext test-pcontext)
-    (cond
-      [(= (length pts+exs) (+ (*num-points*) (*reeval-pts*)))
-        ; got the expected amount of points
-        ; will partition into training and testing set
-        (split-pcontext joint-pcontext (*num-points*) (*reeval-pts*))]
-      [else
-        ; the training set will just be up to the first 256
-        ; the testing set will just be the entire set
-        (define training-count (min 256 (length pts+exs)))
-        (define-values (train-pcontext _)
-          (split-pcontext joint-pcontext training-count (- (length pts+exs) training-count)))
-        (values train-pcontext joint-pcontext)]))
-
-  (values joint-pcontext train-pcontext test-pcontext))
-
 ;; Given a test and a sample of points, returns the test points.
 (define (get-sample test)
-  (define output-repr (test-output-repr test))
-  (define context (test-context test))
-  (*needed-reprs* (list output-repr (get-representation 'bool)))
-
+  (setup-backend! test)
+  (define repr (test-output-repr test))
   (match-define (cons domain-stats joint-pcontext)
     (parameterize ([*num-points* (+ (*num-points*) (*reeval-pts*))])
-      (setup-context!
-       (test-vars test)
-       (or (test-spec test) (test-input test)) (test-pre test)
-       output-repr)))
+      (setup-context! (test-vars test)
+                      (or (test-spec test) (test-input test))
+                      (test-pre test)
+                      repr)))
 
   (define-values (train-pcontext test-pcontext)
     (split-pcontext joint-pcontext (*num-points*) (*reeval-pts*))) 
-
   (for/list ([(pt ex) (in-pcontext test-pcontext)])
     (list pt ex)))
 
 ;; Given a test and a sample of points, computes the error at each point.
 ;; If the sample contains the expected number of points, i.e., `(*num-points*) + (*reeval-pts*)`,
 ;; then the first `*num-points*` will be discarded and the rest will be used for evaluation,
 ;; otherwise the entire set is used.
-(define (get-errors test pts+exs #:seed [seed #f] #:profile [profile? #f])
-  (define output-repr (test-output-repr test))
-  (*context* (test-context test))
-  (*needed-reprs* (list output-repr (get-representation 'bool)))
-  (generate-prec-rewrites (test-conversions test))
+(define (get-errors test pcontext)
+  (setup-backend! test)
+  (unless pcontext
+    (error 'get-errors "cannnot run without a pcontext"))
```

Comment:
Can we move this part to the generic `run-herbie` method?

Reply:
I'm not entirely sure how to support optional arguments to backends. Some backends need a pcontext, other don't. Letting the backend throw the error was the easiest solution.

Reply:
Fine, whatever.